tasks/main.yml

```diff
@@ -15,32 +15,61 @@
     repo: '{{ docker__upstream_repository }}'
     state: 'present'
     update_cache: True
+  register: docker__register_apt_repo
   when: docker__upstream|d() | bool
 
+- name: Make sure that systemd directory exists
+  file:
+    path: '/etc/systemd/system'
+    state: 'directory'
+    owner: 'root'
+    group: 'root'
+    mode: '0755'
+  when: ansible_service_mgr == 'systemd'
+  tags: [ 'role::docker:config' ]
+
+- name: Make sure that docker.service.d directory exists
+  file:
+    path: '/etc/systemd/system/docker.service.d'
+    state: 'directory'
+    owner: 'root'
+    group: 'root'
+    mode: '0755'
+  when: (ansible_service_mgr == 'systemd' and
+        (docker__env_http_proxy is defined or docker__env_https_proxy is defined))
+  tags: [ 'role::docker:config' ]
+
 - name: Remove other version if upstream is modified
   apt:
     name: '{{ item }}'
     state: 'absent'
     install_recommends: False
   register: docker__register_other_version_removed
   with_flattened:
-    - '{{ "docker.io" if docker__upstream|d() else "docker-engine" }}'
+    - '{{ "docker.io" if docker__upstream|d()
+                else docker__packagename }}'
```

Comment:
docker-engine will be dropped/replaced with docker-ce or docker-ee. ATM with the current debops.docker role you can still use it and it installs the new packages (docker-ce/docker-ee) under the old docker-engine name.